api/src/pcapi/scripts/correct_deposits_exp_date/main.py

```diff
@@ -0,0 +1,67 @@
+import argparse
+from datetime import datetime
+import logging
+
+import sqlalchemy as sa
+
+from pcapi.app import app
+from pcapi.core.finance.enum import DepositType
+from pcapi.core.finance.models import Deposit
+from pcapi.core.history.models import ActionHistory
+from pcapi.core.history.models import ActionType
+from pcapi.models import db
+
+
+logger = logging.getLogger(__name__)
+
+app.app_context().push()
+
+
+def update_deposits_expiration_dates(start_id: int = 0, bach_size: int = 1000) -> None:
+    query = (
+        Deposit.query.filter(
+            Deposit.type.in_([DepositType.GRANT_15_17, DepositType.GRANT_18]),
+            Deposit.expirationDate > datetime.utcnow(),
+        )
+        .options(sa.orm.selectinload(Deposit.user))
+        .order_by(Deposit.id)
+    )
+
+    while True:
+        deposits = query.filter(Deposit.id >= start_id, Deposit.id < start_id + bach_size)
+        if not deposits.count():
```

Comment:
```suggestion
        deposits = query.filter(Deposit.id >= start_id, Deposit.id < start_id + bach_size).all()
        if not deposits:
```
quitte à faire une requête pour compter, autant avoir les deposits dans une liste pour ne pas les re-requêter